src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4463,11 +4463,21 @@ bool learnFromConditionalAccessOrBoolConstant()
             }
 
             Debug.Assert(!IsConditionalState);
+            if (binary.OperatorKind.IsLifted()
+                && binary.OperatorKind.Operator() is BinaryOperatorKind.GreaterThan or BinaryOperatorKind.GreaterThanOrEqual or BinaryOperatorKind.LessThan or BinaryOperatorKind.LessThanOrEqual)
+            {
+                Debug.Assert(binary.Type.SpecialType == SpecialType.System_Boolean);
```

Comment:
Is this necessarily true, even in error scenarios? Consider testing with user-defined operators that do not return a boolean.

Reply:
I'll add a test, but there's different operator flags for that (`UserDefinedGreaterThan` and such).